### PR TITLE
add explicit check for OSX version to avoid mysterious breakages

### PR DESCRIPTION
**Pull request by wfurt**
This change adds explicit check for OSX version.

On my 10.11 it fails with following message:
./init-tools.sh
error: OSX version 10.11.6 is too old!

bash -x init.sh

I verified that version expands to VERSION=010011006 as expected. 
I run same script on Sierra and init.sh runs expected. 
(version expands to VERSION=010012005)

### CODE REVIEW

File: init-tools.sh [hunk 1]
```diff
@@ -25,6 +25,12 @@ OSName=$(uname -s)
             OS=OSX
             __DOTNET_PKG=dotnet-dev-osx-x64
             ulimit -n 2048
+	    # Format x.y.z as single integer with three digits for each part
```
Comment (mellinoe) [inline init-tools.sh]: nit: Align this with the above block, it looks like it's not indented far enough.

File: init-tools.sh [hunk 2]
```diff
@@ -25,6 +25,12 @@ OSName=$(uname -s)
...
+	    VERSION=`sw_vers -productVersion| sed -e 's/\./ /g' | xargs printf "%03d%03d%03d"`
+	    if [ "$VERSION" -lt 010012000 ]; then
+	        echo error: OSX version `sw_vers -productVersion` is too old!
```
Comment (mellinoe) [inline init-tools.sh]: Can we print out that "10.12" is required, explicitly?

File: Documentation/building/unix-instructions.md [hunk 3]
```diff
@@ -57,6 +57,8 @@ most systems when you install the development packages).
 
 ### macOS
 
+OSX 10.12 or higer is needed to build corefx 2.0.
```
Comment (justinvp) [inline Documentation/building/unix-instructions.md]: OSX => macOS

File: init-tools.sh (hunk 2 above)
Comment (justinvp) [inline init-tools.sh]: OSX => macOS

Comment (wfurt) [inline init-tools.sh]: The alignment was caused by use of tab. I fixed my profile and I use only spaces now. 
Updated message looks like this now:
error: OSX version 10.11.6 is too old. 10.12 is needed as minimum.

**Comment by wfurt** [comment]
updated based on comments. Text for macOS will need to be still updated more. It is still referencing old - no longer supported versions. I'm running additional tests now.

**Comment by mellinoe** [comment]
The current version looks good to me.

**Pull request merged**